### PR TITLE
add python solution for usaco: cow at large (790)

### PR DESCRIPTION
**Pull request by Amaryllidaceae**
_If any of the below don't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [X] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [X] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making the changes.

### CODE REVIEW

File: solutions/usaco-790.mdx [hunk 1]
```diff
@@ -119,9 +122,62 @@ int main(){
 ```
 </CPPSection>
 
-<JavaSection>
+<PySection>
+```py
+from collections import deque
+
+INF = float('inf')
```
Comment (SansPapyrus683) [inline solutions/usaco-790.mdx:129]: just use this directly

File: solutions/usaco-790.mdx [hunk 2]
```diff
@@ -119,9 +122,62 @@ int main(){
...
+
+    for adj in graph[curr]:
+        if dist_farmer[curr] + 1 < dist_farmer[adj]:
+            dist_farmer[adj] = dist_farmer[curr] + 1
+            queue.appendleft(adj)
+
+# BFS from node k to trace all of Bessie's paths. Increment the farmers needed
+# when dist_bessie[i] = dist_farmer[i], where the additional farmer will cover
+# all of Bessie's escape paths containing node i
+farmers_needed = 0
```
Comment (SansPapyrus683) [inline solutions/usaco-790.mdx:161]: use a multiline comment

File: solutions/usaco-790.mdx [hunk 3]
```diff
@@ -119,9 +122,62 @@ int main(){
...
+queue = deque([k - 1])
+dist_bessie[k - 1] = 0
+while queue:
```
Comment (SansPapyrus683) [inline solutions/usaco-790.mdx:164]: indent w/ tabs

File: solutions/usaco-790.mdx [hunk 4]
```diff
@@ -119,9 +122,62 @@ int main(){
...
+
+with open('atlarge.in', 'r') as infile:
+    n, k = map(int, infile.readline().split())
+    graph = [[] for _ in range(n)]
+    for _ in range(n - 1):
+        f, t = map(lambda i: int(i) - 1, infile.readline().split())
+        graph[f].append(t)
+        graph[t].append(f)
+
+exits = [node for node, adj_list in enumerate(graph) if len(adj_list) == 1]
```
Comment (SansPapyrus683) [inline solutions/usaco-790.mdx:139]: what does this do?

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
yeah this is nice

**Review by SansPapyrus683 — APPROVED** [review]
@ChuangSheep not sure if you know python?

**Comment by ChuangSheep** [comment]
> @ChuangSheep not sure if you know python?

let's say I know the basic but have never used it in contest. Since I rarely use it, Id suggest someone else to do the review here

**Comment by SansPapyrus683** [comment]
@ryanchou-dev

File: solutions/usaco-790.mdx [hunk 5]
```diff
@@ -119,9 +122,66 @@ int main(){
...
+
+# stores minimum steps for a farmer or bessie to reach the node of index = node
+dist_bessie = [float('inf')] * n
+dist_farmer = [float('inf')] * n
+
+"""
+Multi-source BFS from all the exits to calculate, for all nodes, the minimum
+number of steps needed for a farmer to reach that node
+"""
+queue = deque(exits)
```
[the remark this answers is not in the document]
Comment (Amaryllidaceae) [reply solutions/usaco-790.mdx:148]: it's a queue for bfs. not really sure how it can be more descriptive

[the remark this answers is not in the document]
Comment (SansPapyrus683) [reply solutions/usaco-790.mdx:148]: i name it `frontier` in my solutions lol, up to you though

[the remark this answers is not in the document]
Comment (Amaryllidaceae) [reply solutions/usaco-790.mdx:148]: shouldn't queue be fine though? I guess frontier also makes sense but all the c++ solutions I've seen just use q

**Comment by Amaryllidaceae** [comment]
thanks :)

**Pull request merged**